src/notification.c

```diff
@@ -69,62 +96,61 @@ free_notification_xml(struct notification_xml *nxml)
 static void
 check_state(struct notification_xml *nxml)
 {
+	struct delta_item *d;
+	int serial_counter = 0;
+	int serial_diff;
+
 	/* Already have an error or already up to date keep it persistent */
-	if (nxml->state == NOTIFICATION_STATE_ERROR || nxml->state == NOTIFICATION_STATE_NONE)
+	if (nxml->state == NOTIFICATION_STATE_ERROR ||
+	    nxml->state == NOTIFICATION_STATE_NONE)
 		return;
 
 	/* No current data have to go from the snapshot */
-	if (!nxml->current_session_id ||
-	    !nxml->current_serial) {
+	if (nxml->current_session_id == NULL || nxml->current_serial == 0) {
 		nxml->state = NOTIFICATION_STATE_SNAPSHOT;
 		return;
 	}
 
-	if (nxml->session_id && nxml->serial) {
```

Comment:
worried we don't check if we can use session_id and serial now. belts and suspenders kinda thinking.

Reply:
I decided to take this away since the code above does this check:
```
        if (nxml->current_session_id == NULL || nxml->current_serial == 0) {
                nxml->state = NOTIFICATION_STATE_SNAPSHOT;
                return;
        }
```
So the if statement is always true and I decided that less indentation is better than belts and suspenders.

Reply:
difference between "current_" and not. (it is checked outside the only times this function is called but since there is no guarantee of where this function is called...) Fixed it in some of my latest changes.